cli/dcoscli/package/main.py

```diff
@@ -517,3 +525,205 @@ def _uninstall(package_name, remove_all, app_id, cli, app):
+            '{}-{}-{}.zip'.format(
+                package_json['name'],
+                package_json['version'],
+                _hashfile(temp_file.name)))
+
+        if os.path.exists(zip_file_name):
+            raise DCOSException(
+                'Output file [{}] already exists'.format(
+                    zip_file_name))
+
+        # rename with digest
+        util.sh_copy(temp_file.name, zip_file_name)
+
+    # Print the full path to the file
+    emitter.publish(
+        errors.DefaultError(
+            'Created DCOS Universe package [{}].'.format(zip_file_name)))
+
+    return 0
+
+
+def _validate_json_file(fullpath):
+    """Validates the content of the file against its schema. Throws an
+    exception if the file is not valid.
+
+    :param fullpath: full path to the file.
+    :type fullpath: str
+    :return: json object if it is a special file
+    :rtype: dict
+    """
+
+    if os.path.basename(fullpath) == 'command.json':
+        schema_path = 'data/universe-schema/command.json'
+    elif os.path.basename(fullpath) == 'config.json':
+        schema_path = 'data/universe-schema/config.json'
+    elif os.path.basename(fullpath) == 'package.json':
+        schema_path = 'data/universe-schema/package.json'
+    else:
+        raise DCOSException(('Error while process package directory. Unknown '
+                             'file: {}').format(fullpath))
+
+    special_schema = util.load_jsons(
+        pkg_resources.resource_string('dcoscli', schema_path).decode('utf-8'))
+
+    with util.open_file(fullpath) as special_file:
+        special_json = util.load_json(special_file)
+
+    errs = util.validate_json(special_json, special_schema)
+    if errs:
+        emitter.publish(
+            errors.DefaultError(
+                'Error validating JSON file [{}]'.format(fullpath)))
+        raise DCOSException(util.list_to_err(errs))
+
+    return special_json
+
+
+def _hashfile(filename):
+    """Calculates the sha256 of a file
+
+    :param filename: path to the file to sum
+    :type filename: str
+    :returns: digest in hexadecimal
+    :rtype: str
+    """
+
+    hasher = hashlib.sha256()
+    with open(filename, 'rb') as f:
+        for chunk in iter(lambda: f.read(4096), b''):
+            hasher.update(chunk)
+    return hasher.hexdigest()
+
+
+def _import_assets(assets_directory, zip_file):
+    """Import the assets directory
+
+    :param assets_directory: path to the assets directory
+    :type assets_directory: str
+    :param zip_file: zip file object
+    :type zip_file: zipfile.ZipFile
+    :rtype: None
+    """
+
+    for filename in sorted(os.listdir(assets_directory)):
+        fullpath = os.path.join(assets_directory, filename)
+        if filename == 'uris' and os.path.isdir(fullpath):
+            _import_uris(fullpath, zip_file)
+        else:
+            # anything else is an error
+            raise DCOSException(
+                'Unknown file [{}]'.format(fullpath))
```

Comment:
Hmm. Maybe we should print: "Error bundling. Extra file in package directory [...]".